src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -477,6 +477,8 @@ private function parseFileToDOM(string $file): \DOMDocument
                             continue;
                         }
                         if ('provider' === $parent->localName || 'firewall' === $parent->localName) {
+                            trigger_deprecation('symfony/security-bundle', '7.2', 'Custom %s must now be namespaced; please update your security configuration "%s" tag.', 'provider' === $parent->localName ? 'providers' : 'authenticators', $tagName);
```

Comment:
How is the code going to look like once clean things for SF8?
Any way to hint about this already somewhere?

Reply:
We’ll have to revert the changes made by https://github.com/symfony/symfony/pull/57493 to `XmlFileLoader::parseFileToDOM` so that the whole `catch` block is just

```php
throw new InvalidArgumentException(\sprintf('Unable to parse file "%s": ', $file).$e->getMessage(), $e->getCode(), $e);
```
again. Not sure if or how I can convey that in the code :thinking:

Reply:
can you add that line as a comment at the beginning of the catch block maybe, with a small comment above?

Reply:
Sorry, got sidetracked. Iʼll be able to push it tomorrow but feel free to take over this branch in the meantime.

Reply:
Done in https://github.com/symfony/symfony/pull/58083/commits/6fcc591e0a6f47a3d4bc81fbcaf383ac4fd94044